src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8406,7 +8412,7 @@ private FixedStatementSyntax ParseFixedStatement(SyntaxList<AttributeListSyntax>
 
             var saveTerm = _termState;
             _termState |= TerminatorState.IsEndOfFixedStatement;
-            var decl = ParseVariableDeclaration();
```

Comment:
renamed this to make it clear which caller contexts it is expected to be used in.  The old name was highly misleading and could make people think they should they should use this in any circumstance they want to parse a variable-declaration (like in a field for example).

Reply:
importantly, all of these are the constructs that *do* want to stop parsing the variable declaration once they hit an errant close paren, as their construct themselves want to consume the close paren into themselves.